### DISCUSSION TITLE
Function of units in deviceResource properties

### DISCUSSION THREAD

**Discussion by sounak-1989** [edgexfoundry/community#32] [Q&A]
Hi 
I wanted to know the use of the property "units" under deviceResources in a profile. I don't see any way to access that unit from code or to send it back as a part of the response. Let me know how the "unit" property can be utilized.

Thanks
Sounak

**Comment by cloudxxx8** [comment]
In the Device Service implementation, you can query the device resource via the SDK DS API, and you can access the unit from it
https://github.com/edgexfoundry/device-sdk-go/blob/7d510a9be1a3cd2158385898861d08f1c1b53357/pkg/interfaces/service.go#L78

Also, the readings sent out from Device Service should contain an unit field if the config `Writable.Reading.ReadingUnits = true`
https://github.com/edgexfoundry/device-sdk-go/blob/7d510a9be1a3cd2158385898861d08f1c1b53357/example/cmd/device-simple/res/configuration.toml#L14

**Reply by sounak-1989** [reply]
@cloudxxx8 thanks a lot for the references. Helped me a lot.
